perfdata/type.go

```diff
@@ -51,7 +86,10 @@ type Perfdata struct {
 }
 
 // String returns the proper format for the plugin output
-func (p Perfdata) String() string {
+// Returns an eror in some known cases where the value of a data type does not
+// represent a valid measurement, see the explanation for "formatNumeric" for
+// perfdata values.
+func (p Perfdata) String() (string, error) {
```

Comment:
I don't like that this is no longer implements the Stringer interface. https://pkg.go.dev/fmt#Stringer

I rather have the formatNumeric handle that and the Stringer interface be implemented, since it is more expectable when having a String() method on a struct.

Not sure how else we could or should communicate that NaN/Inf are not printed. Maybe just docs?

Reply:
I see your point. I think I have an idea.

Reply:
I restored the behaviour of `String` by using the new functionality and just ignore errors.

Reply:
Coolio. I'll have a look later.